base/pmap.jl

```diff
@@ -1,8 +1,12 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+type BatchProcessingError <: Exception
+    data
+    ex
+end
 
 """
-    pgenerate([::WorkerPool], f, c...) -> iterator
+    pgenerate([::WorkerPool], f, c...) -> (iterator, process_batch_errors)
```

Comment:
I would very much prefer it if `pgenerate` just returned an iterator.
If there is some error batch processing that must be done, it should be handled either by a wrapper function around `f`, or if that is not possible, by the mechanics of `pgenerate` inside `next`.

This change seems to preclude infinite iterators.
One of my primary motivations for having an async/parallel mapper/generator is to be able to chain together a series of generators to an infinite source of work. e.g. https://github.com/JuliaLang/julia/issues/14515#issuecomment-169536524

`pmap` is finite, so I guess there is no fundamental reason not to do clean-up-errors-at-the-end there.
However, to me it is important that `pgenerate` should handle infinite input and should just return an iterator.